modin/core/storage_formats/base/query_compiler.py

```diff
@@ -3089,7 +3089,12 @@ def write_items(df, broadcasted_items):
             if isinstance(df.iloc[row_numeric_index, col_numeric_index], pandas.Series):
                 broadcasted_items = broadcasted_items.squeeze()
             df.iloc[
-                list(row_numeric_index), list(col_numeric_index)
+                list(row_numeric_index)
+                if not isinstance(row_numeric_index, slice)
+                else row_numeric_index,
+                list(col_numeric_index)
+                if not isinstance(col_numeric_index, slice)
+                else col_numeric_index,
```

Comment:
This is done for cases where `slice(None)` is passed in `write_items`.

Reply:
```python
if is_iterable(row_numeric_index):
    row_numeric_index = list(row_numeric_index)
if is_iterable(col_numeric_index):
    col_numeric_index = list(col_numeric_index)
df.iloc[row_numeric_index, col_numeric_index] = broadcasted_items
```
maybe something like this? The present ternary operators look pretty heavy to read :)